File: test_live.py
import requests
import json

url = "https://choir-satb-extractor.onrender.com/analyze"
print("Uploading test file to the live Render backend...")

# We need a dummy midi file to upload. 
# Let's create a tiny one.
from music21 import stream, note, midi
s = stream.Stream()
s.append(note.Note('C4'))
s.write('midi', fp='test.mid')

files = {'file': ('test.mid', open('test.mid', 'rb'), 'audio/midi')}
response = requests.post(url, files=files)

print("Status Code:", response.status_code)
try:
    print(json.dumps(response.json(), indent=2))
except:
    print("Response Text:", response.text)

File: test_live.js
const fs = require('fs');

async function runTest() {
  try {
    const formData = new FormData();
    // Assuming test.mid exists
    const fileBlob = new Blob([fs.readFileSync('test.mid')], { type: 'audio/midi' });
    formData.append('file', fileBlob, 'test.mid');
    
    console.log("Sending POST to https://choir-satb-extractor.onrender.com/analyze ...");
    const response = await fetch('https://choir-satb-extractor.onrender.com/analyze', {
      method: 'POST',
      body: formData
    });
    
    console.log(`Status: ${response.status} ${response.statusText}`);
    const text = await response.text();
    try {
      console.log(JSON.stringify(JSON.parse(text), null, 2));
    } catch (e) {
      console.log("Raw Response (not JSON):", text);
    }
  } catch (error) {
    console.error("Fetch failed:", error);
  }
}
runTest();
